### PR TITLE
Sunburst

### PR DESCRIPTION
**Pull request by mcnuttandrew**
sunburst

Add a top level chart type, sunbursts. This chart makes use of the recently created arcSeries to handle all of the data processing for it. Still to do

- [ ] Tests
- [ ] Docs
- [ ] JSdocs Where appropriate
- [ ] Animation is currently broken, the interpolator doesn't know how to address deeply nested data!

The API of this chart is the same as the treemap, so it should be reasonably easy for users to start using it right away! Would love comments if anyone has any

### CODE REVIEW

File: showcase/sunbursts/basic-sunburst.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  const name = node.data && node.data.name || node.name || '';
+  if (!node.parent) {
+    return 'root';
+  }
+
+  return `${getTextPath(node.parent)} > ${name}`;
+}
+
+function getKeyPath(node) {
+  // const name = node.data && node.data.name || node.name || '';
```
Comment (javidhsueh) [inline showcase/sunbursts/basic-sunburst.js]: not used?

File: showcase/sunbursts/basic-sunburst.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+function decorateData(data) {
+  if (!data.children) {
+    return data;
+  }
+  // add a fill to all the uncolored cells
+  if (!data.color) {
+    data.style = {
+      fill: '#223F9A'
```
Comment (javidhsueh) [inline showcase/sunbursts/basic-sunburst.js]: should make a constant for this magic color

File: showcase/sunbursts/basic-sunburst.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      <div className="basic-sunburst-example-wrapper">
+        <Sunburst
+          animation
+          className="basic-sunburst-example"
+          onValueMouseOver={node => {
+            const yy = getTextPath(node.parent);
+            const xx = getKeyPath(node.parent);
+            // console.log(xx)
+            // console.log(this)
+            // debugger;
```
Comment (javidhsueh) [inline showcase/sunbursts/basic-sunburst.js]: not used

File: showcase/sunbursts/basic-sunburst.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            this.setState({value: yy});
+          }}
+          onValueClick={node => {
+            this.setState({subState: subState ? false : node});
+          }}
+          style={{
```
Comment (javidhsueh) [inline showcase/sunbursts/basic-sunburst.js]: it should be a const 'defaultStyle'

File: src/plot/series/arc-series.js [hunk 5]
```diff
@@ -142,20 +140,21 @@ class ArcSeries extends AbstractSeries {
         ref="container"
         transform={`translate(${marginLeft + xTranslate},${marginTop + yTranslate})`}>
         {data.map((row, i) => {
+          const cleanedRow = {...row};
```
Comment (javidhsueh) [inline src/plot/series/arc-series.js]: why we need to shallow copy the row?

Comment (javidhsueh) [inline src/plot/series/arc-series.js]: maybe add more comments?

File: src/sunburst/index.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+    return partitionFunction(structuredInput).descendants()
+      .reduce((res, cell, index) => {
+        if (hideRootNode && index === 0) {
+          return res;
+        }
+
+        return res.concat([{
+          angle0: Math.max(0, Math.min(2 * Math.PI, x(cell.x0))),
+          angle: Math.max(0, Math.min(2 * Math.PI, x(cell.x1))),
```
Comment (javidhsueh) [inline src/sunburst/index.js]: shouldn't it be called angle1 by following the same naming convention?

File: src/sunburst/index.js [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+          radius0: Math.max(0, y(cell.y0)),
+          radius: Math.max(0, y(cell.y1)),
```
Comment (javidhsueh) [inline src/sunburst/index.js]: radius1 instead?

File: src/sunburst/index.js [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+  }
+  return {
+    ...node,
+    parent: cleanParent(node.parent)
+  };
+}
+
+class Sunburst extends React.Component {
+
+  static get propTypes() {
```
Comment (javidhsueh) [inline src/sunburst/index.js]: colorType is missing

File: src/plot/xy-plot.js [hunk 9]
```diff
@@ -46,6 +46,28 @@ const DEFAULT_MARGINS = {
   bottom: 40
 };
 
+function cleanseData(data) {
+  return data.map(series => {
+    if (!Array.isArray(series)) {
+      return series;
+    }
+    return series.map(row => ({...row, parent: null}));
```
Comment (javidhsueh) [inline src/plot/xy-plot.js]: it seems to be dealing with sunburst case. Can we just have a white list of the props instead?

File: showcase/index.js [hunk 10]
```diff
@@ -107,6 +109,8 @@ export const showCase = {
   SimpleTreemap,
   TreemapExample,
 
+  BasicSunburst,
```
Comment (javidhsueh) [inline showcase/index.js]: alphabetize all showcases?

File: src/plot/xy-plot.js (hunk 9 above)
Comment (mcnuttandrew) [inline src/plot/xy-plot.js]: i also want to re-write treemap to be in XY coordinates, so itll also be dealing with that.

File: src/sunburst/index.js (hunk 7 above)
Comment (mcnuttandrew) [inline src/sunburst/index.js]: that'd be following the d3 naming convention, not the react-vis one. There is not a notion of {attr}1, there is only {attr} and {attr}0.

File: showcase/index.js (hunk 10 above)
Comment (mcnuttandrew) [inline showcase/index.js]: right now they are broken apart by group, but that is definately something to do in the future, specifically i think itll be addressed in #343

**Pull request merged**